### PR TITLE
Markdown Reporting Index File

### PR DESCRIPTION
**Pull request by xm1k3**
## Proposed changes

image

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [ ] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: v2/pkg/reporting/exporters/markdown/markdown.go [hunk 1]
```diff
@@ -66,6 +66,34 @@ func (exporter *Exporter) Export(event *output.ResultEvent) error {
 	dataBuilder.WriteString(description)
 	data := dataBuilder.Bytes()
 
+	// index generation
+
```
Comment (Mzack9999) [inline v2/pkg/reporting/exporters/markdown/markdown.go:70]: please remove the extra new line

File: v2/pkg/reporting/exporters/markdown/markdown.go [hunk 2]
```diff
@@ -66,6 +66,34 @@ func (exporter *Exporter) Export(event *output.ResultEvent) error {
...
+	file, err := os.OpenFile(filepath.Join(exporter.directory, "index.md"), os.O_CREATE|os.O_WRONLY, 0644)
```
Comment (Mzack9999) [inline v2/pkg/reporting/exporters/markdown/markdown.go:71]: Why the file index header is written over and over? Shouldn't the index creation happen in the `New(..)`?

File: v2/pkg/reporting/exporters/markdown/markdown.go [hunk 3]
```diff
@@ -66,6 +66,34 @@ func (exporter *Exporter) Export(event *output.ResultEvent) error {
...
+	_, err = file.WriteString("|Hostname/IP|Finding|Severity|\n")
+	if err != nil {
+		return err
+	}
+	_, err = file.WriteString("|-|-|-|\n")
+	if err != nil {
+		return err
+	}
+
+	file, err = os.OpenFile(filepath.Join(exporter.directory, "index.md"), os.O_APPEND|os.O_WRONLY, 0644)
```
Comment (Mzack9999) [inline v2/pkg/reporting/exporters/markdown/markdown.go:86]: The variable is still referenced at line 75 (`defer file.Close()`)

File: v2/pkg/reporting/exporters/markdown/markdown.go [hunk 4]
```diff
@@ -66,6 +66,34 @@ func (exporter *Exporter) Export(event *output.ResultEvent) error {
...
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	_, err = file.WriteString("|" + event.Host + "|" + event.Info.Name + "|" + event.Info.SeverityHolder.Severity.String() + "|\n")
```
Comment (Mzack9999) [inline v2/pkg/reporting/exporters/markdown/markdown.go:92]: `event.Host` as per GH issue description should be a hyperlink to the specific markdown file containing the corresponding issue detail.

**Review by Mzack9999 — CHANGES_REQUESTED** [review]

File: v2/pkg/reporting/exporters/markdown/markdown.go [hunk 5]
```diff
@@ -11,6 +11,8 @@ import (
 	stringsutil "github.com/projectdiscovery/utils/strings"
 )
 
+var indexFileName = "index.md"
```
Comment (Mzack9999) [inline v2/pkg/reporting/exporters/markdown/markdown.go:14]: `const` can be used as well

File: v2/pkg/reporting/exporters/markdown/markdown.go [hunk 6]
```diff
@@ -33,6 +35,22 @@ func New(options *Options) (*Exporter, error) {
 		directory = dir
 	}
 	_ = os.MkdirAll(directory, 0755)
+
+	// index generation header
+	file, err := os.OpenFile(filepath.Join(directory, indexFileName), os.O_CREATE|os.O_WRONLY, 0644)
```
Comment (Mzack9999) [inline v2/pkg/reporting/exporters/markdown/markdown.go:40]: This block can be replaced with `os.WriteFile` (as in line 99)

**Review by Mzack9999 — CHANGES_REQUESTED** [review]

**Review by Mzack9999 — APPROVED** [review]

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**